chain/chunks/src/lib.rs

```diff
@@ -775,7 +779,11 @@ impl ShardsManager {
 
     /// Returns true if transaction is not in the pool before call
     pub fn insert_transaction(&mut self, shard_id: ShardId, tx: SignedTransaction) -> bool {
-        self.tx_pools.entry(shard_id).or_insert_with(TransactionPool::new).insert_transaction(tx)
+        let seed = self.random_seed(shard_id);
+        self.tx_pools
+            .entry(shard_id)
+            .or_insert_with(|| TransactionPool::new(seed))
+            .insert_transaction(tx)
```

Comment:
This whole PR is full of "why"s to me.

Why do we need to generate a random seed on every insertion of a transaction?

Why `[3; 32]` and `[4; 32]`? (All magical consts should have names to understand their nature)

Why `random_seed` below is implemented that way? (why 17 and 23? why `shard_id / 256` - I feel it is going to be 0 most of the time)

Reply:
yeah same question. Also this may have nontrivial impact on the performance of `insert_transaction`.